overlord/hookstate/hookmgr.go

```diff
@@ -295,6 +318,19 @@ func runHookAndWait(snapName string, revision snap.Revision, hookName, hookConte
 		return nil, err
 	}
 
+	// add timeout handling
+	var killTimer *time.Timer
+	if maxRuntime > 0 {
+		killTimer = time.AfterFunc(maxRuntime, func() {
```

Comment:
sorry, I didn't read this carefully enough last night,   I think here instead of using the timer callback we can use the timer channel

    var killTimerCh <-chan Time
    if maxRunTime > 0 {
        killTimer := time.NewTimer(...)
        defer ...
        killTimerCh = killTimer.C
    }

then below in the select:

    case <- tomb.Dying():
    case <- killTimerCh:
    }
    // abort code goes here...